src/Chapter2.hs

```diff
@@ -336,7 +360,9 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList start fin xs = if start < 0 || fin < 0 then [] else
+  let len = fin - start + 1
+  in take len $ drop start xs
```

Comment:
Your implementation works correctly when `start > fin` because of how `take` and `drop` implemented, but I think you can add one more extra case in here for more efficient implementation.

Reply:
What should be the expected behavior when `start < fin`? My intuition tells me it should return empty list. What do you think? My implementation does return empty list for this case. What do you mean by more efficient? Like I should explicitly check for `start > fin` and return empty list before doing the more generic pattern matching?

Reply:
What would you suggest? Doing something like `if start > fin [] else ....`?

Reply:
You can just add ` || start > fin` in your first check inside `if-then-else`

Reply:
👍